test/bootloader_test.rb

```diff
@@ -150,4 +158,177 @@
+        expect(Yast::BootCommon.changed).to eq(false)
+      end
+    end
+
+    context "when no target is specified" do
+      it "uses :common by default" do
+        subject.modify_kernel_params(params)
+
+        expect(Yast::BootCommon.globals)
+          .to eq(kernel_line(:common) => append, "__modified" => "1")
+        expect(Yast::BootCommon.changed).to eq(true)
+      end
+    end
+
+    context "when a target is specified" do
+      it "adds parameter for that target" do
+        subject.modify_kernel_params(:xen_guest, params)
+
+        expect(Yast::BootCommon.globals)
+          .to eq(kernel_line(:xen_guest) => append, "__modified" => "1")
+        expect(Yast::BootCommon.changed).to eq(true)
+      end
+    end
+
+    context "when multiple targets are specified" do
+      it "adds parameters to each target" do
+        subject.modify_kernel_params(:xen_host, :xen_guest, params)
+
+        expect(Yast::BootCommon.globals)
+          .to eq(kernel_line(:xen_host) => append,
+                 kernel_line(:xen_guest) => append,
+                 "__modified" => "1")
+        expect(Yast::BootCommon.changed).to eq(true)
+      end
+    end
+
+    context "when targets are specified as an array" do
+      it "adds parameters to each target" do
+        subject.modify_kernel_params([:xen_host, :xen_guest], params)
+
+        expect(Yast::BootCommon.globals)
+          .to eq(kernel_line(:xen_host) => append,
+                 kernel_line(:xen_guest) => append,
+                 "__modified" => "1")
+        expect(Yast::BootCommon.changed).to eq(true)
+      end
+    end
+
+    context "when a parameter is set to be removed" do
+      let(:initial_lines) { { kernel_line(:common) => "quiet #{append}" } }
+      let(:params) { { "quiet" => :missing } }
+
+      it "removes parameter from the given target" do
+        subject.modify_kernel_params(:common, params)
+
+        expect(Yast::BootCommon.globals)
+          .to eq(kernel_line(:common) => append, "__modified" => "1")
+        expect(Yast::BootCommon.changed).to eq(true)
+      end
+    end
+
+    context "when a parameter is set to be just present" do
+      let(:params) { { "quiet" => :present } }
+
+      it "adds the parameter to the given target without any value" do
+        subject.modify_kernel_params(:common, params)
+
+        expect(Yast::BootCommon.globals)
+          .to eq(kernel_line(:common) => "quiet", "__modified" => "1")
+        expect(Yast::BootCommon.changed).to eq(true)
+      end
+    end
+
+    context "when parameter is 'vga'" do
+      let(:params) { { "vga" => "80" } }
+
+      it "adds the parameter as 'vgamode' to the global scope and does not mark BootCommon as 'modified'" do
+        subject.modify_kernel_params(:common, params)
+        expect(Yast::BootCommon.globals)
+          .to eq("vgamode" => "80")
+        expect(Yast::BootCommon.changed).to eq(false)
+      end
+    end
+
+    context "when parameter is 'root' (unsupported)" do
```

Comment:
root parameter is supported, just grub2 do not allow modification as it is generated on its own. So changing it have no effect.